Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -832,3 +860,141 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Replicate Encrypted Properties wit
 }
 #endif //#ifdef COUCHBASE_ENTERPRISE
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Remove Doc From Channel SG", "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    // one collection now now. Will use multiple collection when SG is ready.
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+        Roses
+    };
+    std::array<C4Collection*, collectionCount> collections =
+        collectionPreamble(collectionSpecs, "sguser", "password");
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+
+    string        doc1ID {idPrefix + "doc1"};
+    vector<string> chIDs {idPrefix+"a", idPrefix+"b"};
+
+    C4Error error;
+    DEFER {
+        REQUIRE(assignUserChannel(this, {"*"}, &error));
```

Comment:
I am not sure this will be allowed by C++ rules.  This is equivalent to potentially throwing an exception in a destructor.

Reply:
A good point. It would terminate if hit. On the other point, this is kind of fatal, because it affects the SG user. All tests that works with SG will be affected. This is a disadvantage of "isolation by channel", as compared with "isolation by doc IDs," because we are fiddling with global attributes that affect all SG tests.

Reply:
Removed REQUIRE.